### ISSUE TITLE
Improve `FileRegistry` creation APIs

### ISSUE THREAD

**Issue by ematipico** [getgrit/gritql#591]
At the moment, [`FileRegistry::new_from_paths`](https://github.com/getgrit/gritql/blob/50b976db637ecbd60daee3f59cc4f868f59bccb0/crates/grit-pattern-matcher/src/pattern/state.rs#L97-L105) has unpleasant signature.

Creating a `Vec<&'a PathBuf>` can be inefficient because it forces consumers to create a vector **of references**, and/or it could be difficult to pass `std` path if the consumers rely on other libraries to handle paths, like [`camino`](https://docs.rs/camino/latest/camino/).

Suggestion: change the API signature to `&[&Path]`, or pass an iterator. 

cc @arendjr

**Comment by morgante** [comment]
Agreed that an iterator would make sense, since we're calling iter() inside anyways. Happy to review a PR.
